back-end/src/main/java/server/UpdateRequestHandler.java

```diff
@@ -0,0 +1,179 @@
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+import javax.net.ssl.HttpsURLConnection;
+import main.Config;
+import main.DbConnector;
+import main.Main;
+
+public class UpdateRequestHandler implements HttpHandler {
+    
+    private final Gson gson;
+
+    public UpdateRequestHandler() {
+        gson = new GsonBuilder().create();
+    }
+
+    @Override
+    public void handle(HttpExchange httpExchange) throws IOException {
+        System.out.println("Serving /update");
+        try {
+            String response;
+            String canvasId;
+            String layerId;
+            String keyColumn;
+
+            // should be a POST request
+            if (!httpExchange.getRequestMethod().equalsIgnoreCase("POST")) {
+                Server.sendResponse(httpExchange, HttpsURLConnection.HTTP_BAD_METHOD, "");
+                return;
+            }
+
+            // get body data of POST request
+            String body = new String(readBody(httpExchange), "utf-8");
+
+            HashMap<String, String> map = new HashMap<String, String>();
+            JSONObject jObject = new JSONObject(body);
+            Iterator<?> keys = jObject.keys();
+            while( keys.hasNext() ){
+                String key = (String)keys.next();
+                String value = jObject.getString(key); 
+                map.put(key, value);
+            }
+
+            canvasId = map.get("canvasId").toString();
+            layerId = map.get("layerId").toString();
+            keyColumn = map.get("primaryKeyColumn").toString();
+            System.out.println("canvas id is: " + canvasId);
+            System.out.println("layer id is: " + layerId);
+            System.out.println("key column is: " + keyColumn);
+            String rawAttributes = map.get("objectAttributes").toString();
+            System.out.println("attributes string ->" + rawAttributes);
+            HashMap<String, String> objectAttrs = new HashMap<String, String>();
+            JSONObject attrJson = new JSONObject(rawAttributes);
+            Iterator<?> attrKeys = attrJson.keys();
+            while ( attrKeys.hasNext() ) {
+                String key = (String)attrKeys.next();
+                String value = attrJson.getString(key);
+                objectAttrs.put(key, value);
+            }
+            System.out.println("object attrs: " + objectAttrs);
+
+            String tableName = "bbox_" + Main.getProject().getName()
+                                     + "_" + canvasId + "layer" + layerId;
+            HashMap<String, String> attrColumnTypes = new HashMap<String, String>();
+            Statement stmt = DbConnector.getStmtByDbName(Config.databaseName);
+            String typeQuery = 
+                    "SELECT column_name, data_type FROM information_schema.columns WHERE table_name = "
+                    + "'" + tableName + "';";
```

Comment:
Nice job Peter for considering different types of fields!

Actually right now everything is typed text (color my face red). Although distinguishing types makes no difference here, let's keep this in case we want to support typed fields in the future.